source/_components/sensor.weightedaverage.markdown

```diff
@@ -0,0 +1,43 @@
+---
+layout: page
+title: "Weighted Average Sensor"
+description: "Instructions on how to add weighted average sensors within Home Assistant."
+date: 2019-06-01 01:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: home-assistant.png
+ha_category:
+  - Sensor
+ha_release: 0.93
+ha_iot_class: Local Push
+ha_qa_scale: internal
+---
+
+The `weightedaverage` sensor calculates an average value from another sensor, that is weighted according to the duration for each individual sensor value.
+This is useful when the source sensor state is updated at uneven intervals, typically a temperature sensor that is only updated when the temperature changes. The weighted average takes into account the 'weight' (i.e. the duration) for each temperature.
+The sensor also records the maximum and minimum sensor values for the given duration, as well as the top rates at which the sensor value has ascended or descended.
+
+
+## {% linkable_title Configuration %}
```

Comment:
Please add also the [configuration](https://developers.home-assistant.io/docs/en/documentation_create_page.html#configuration)